ingestion/src/lib.rs

```diff
@@ -93,26 +101,71 @@ impl FromRef<IngestorState> for Arc<RwLock<(ParamPermitTable, StationPermitTable
     }
 }
 
+impl FromRef<IngestorState> for Arc<rove_connector::Connector> {
+    fn from_ref(state: &IngestorState) -> Arc<rove_connector::Connector> {
+        state.rove_connector.clone()
+    }
+}
+
+impl FromRef<IngestorState> for Arc<HashMap<(i32, RelativeDuration), rove::Pipeline>> {
+    fn from_ref(state: &IngestorState) -> Arc<HashMap<(i32, RelativeDuration), rove::Pipeline>> {
+        state.qc_pipelines.clone()
+    }
+}
+
 /// Represents the different Data types observation can have
 #[derive(Debug, PartialEq)]
 pub enum ObsType<'a> {
     Scalar(f32),
     NonScalar(&'a str),
```

Comment:
Q1: I guess that for scalar obs values there will never be a need for storing with higher precision than f32? And if the need should arise, those should be stored as NonScalar (and we'd accept the performance penalty)?
Q2: Will there be some kind of basic validation of NonScalar observations, such as restricting to some maximum size etc.?

Reply:
1: After investigating the performance implications, we actually decided to switch to f64 everywhere. See #54 for more info.

2: The point is that we don't know anything about the non-scalar parameters apart from that they probably shouldn't be stored as floats, so strings are the safest option to avoid data loss, as they are the form the data is delivered in. As we identify these parameters, we plan to store them in more appropriate types, i.e integer, enum, datetime; I think that's where validation becomes relevant.